store/types.go

```diff
@@ -0,0 +1,67 @@
+package store
+
+import (
+	"github.com/FourthState/plasma-mvp-sidechain/plasma"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	"math/big"
+)
+
+// Wallet holds reference to the total balance, unspent, and spent outputs
+// at a given address
+type Wallet struct {
+	Balance *big.Int          // total amount avaliable to be spent
+	Unspent []plasma.Position // position of unspent transaction outputs
+	Spent   []plasma.Position // position of spent transaction outputs
+}
+
+/* Wrap plasma deposit with spend information */
+type Deposit struct {
+	Deposit   plasma.Deposit
+	Spent     bool
+	SpenderTx []byte // transaction hash that spends this deposit
+}
+
+/* Wrap plasma output with spend information */
+type Output struct {
+	Output    plasma.Output
+	Spent     bool
+	SpenderTx []byte // transaction that spends this output
+}
+
+// TODO: remove
+type OutputInfo struct {
+	Output Output
+	Tx     Transaction
+}
+
+/* Wrap plasma transaction with spend information */
+type Transaction struct {
+	Transaction      plasma.Transaction
+	ConfirmationHash []byte
+	Spent            []bool
+	SpenderTxs       [][]byte // transaction hashes that spend the outputs of this transaction
+	Position         plasma.Position
+}
+
+// TransactionOutput holds all transactional information related to an output
+// It is used to return output information from the store
+type TransactionOutput struct {
```

Comment:
We can rename this or shorten to TxOutput, but I think the name is fairly appropriate since the extra information added comes for the transactional context of the outputs